packages/nocodb/src/modules/jobs/migration-jobs/nc_job_007_recover_order_column.ts

```diff
@@ -0,0 +1,273 @@
+
+      this.log(`Total models to be processed: ${numberOfModelsToBeProcessed}`);
+
+      const queue = new PQueue({ concurrency: PARALLEL_LIMIT });
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        if (queue.pending > PARALLEL_LIMIT * 2) {
+          await new Promise((resolve) => setTimeout(resolve, 1000));
+          continue;
+        }
+
+        this.processingModels = this.processingModels.filter(
+          (m) => m.processing,
+        );
+
+        const models = await ncMeta
+          .knexConnection(MetaTable.MODELS)
+          .select([
+            `${MetaTable.MODELS}.id`,
+            `${MetaTable.MODELS}.source_id`,
+            `${MetaTable.MODELS}.table_name`,
+            `${MetaTable.MODELS}.base_id`,
+            ...(isEE ? [`${MetaTable.MODELS}.fk_workspace_id`] : []),
+          ])
+          .join(
+            MetaTable.SOURCES,
+            `${MetaTable.MODELS}.source_id`,
+            '=',
+            `${MetaTable.SOURCES}.id`,
+          )
+          .where(`${MetaTable.MODELS}.mm`, false)
+          .where((builder) => {
+            builder.where(`${MetaTable.SOURCES}.is_meta`, true);
+            builder.orWhere({ is_local: true });
+          })
+          .where((builder) => {
+            builder
+              .where(`${MetaTable.MODELS}.deleted`, false)
+              .orWhereNull(`${MetaTable.MODELS}.deleted`);
+          })
+          .whereNotIn(
+            `${MetaTable.MODELS}.id`,
+            this.processingModels.map((m) => m.fk_model_id),
+          )
+          .where((builder) => {
+            builder
+              .whereNotExists(function () {
+                this.select(1)
+                  .from(MetaTable.COLUMNS)
+                  .whereRaw(
+                    `${MetaTable.COLUMNS}.fk_model_id = ${MetaTable.MODELS}.id`,
+                  )
+                  .where(`${MetaTable.COLUMNS}.uidt`, 'Order');
+              })
+              .orWhereExists(function () {
+                this.select(1)
+                  .from(MetaTable.COLUMNS)
+                  .whereRaw(
+                    `${MetaTable.COLUMNS}.fk_model_id = ${MetaTable.MODELS}.id`,
+                  )
+                  .where(`${MetaTable.COLUMNS}.uidt`, 'Decimal')
+                  .where(`${MetaTable.COLUMNS}.system`, true)
+                  .where(`${MetaTable.COLUMNS}.column_name`, '=', 'nc_order');
+              });
+          })
+          .limit(PARALLEL_LIMIT * 2);
+
+        if (!models?.length) break;
+
+        for (const model of models) {
+          this.processingModels.push({
+            fk_model_id: model.id,
+            processing: true,
+          });
+
+          queue
+            .add(() => this.processModel(model, ncMeta))
+            .catch((e) => {
+              this.log(`Error processing model ${model.id}:`, e.message);
+            });
+        }
+      }
+
+      await queue.onIdle();
+
+      await ncMeta.disableUpgraderMode();
+
```

Comment:
lets have this in catch block as well (we can do same in 005 migration bc we disable cache on enabling & we need to disable this mode whether the result)